gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/vertx/web/handler/impl/internal/mfa/MfaFilterContext.java

```diff
@@ -131,6 +133,29 @@ public boolean isDeviceRiskAssessmentEnabled() {
     public boolean isChallengeOnceCompleted() {
         return TRUE.equals(session.get(MFA_CHALLENGE_COMPLETED_KEY));
     }
+
+
+    public void setDefaultFactorWhenApplied(RuleEngine ruleEngine) {
+        String enrollingFactorId = session.get(ENROLLED_FACTOR_ID_KEY);
+        FactorSettings factorSettings = client.getFactorSettings();
+        if (nonNull(enrollingFactorId) && nonNull(factorSettings) && !matchFactorRule(enrollingFactorId, factorSettings, ruleEngine)) {
```

Comment:
**issue**: this condition will never been met as the matchFactorRule test if the enrollingFactorId is the default factor